core/src/prettier/mod.rs

```diff
@@ -0,0 +1,119 @@
+mod document;
+mod print;
+
+pub use self::document::{concat, const_text, display, flatten, indent, nl, text, Document};
+
+use alloc::string::String;
+use core::fmt;
+
+pub trait PrettyPrint {
```

Comment:
Could we add some docs to this trait? e.g., summarizing what functionality we get with it.

Reply:
Sure thing, not sure why I haven't done that yet honestly